### PR TITLE
Add compat data MediaStreamTrack

### PR DESCRIPTION
**Pull request by mec**

### CODE REVIEW

**Comment by foolip** [comment]
Came here somewhat at random to take a look what this is gonna look like. The manual table currently in https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack has more information than is preserved here, for example about getSources. What's the plan for that?

File: api/MediaStreamTrack.json [hunk 1]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": false
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": true
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Please change to `true`.

File: api/MediaStreamTrack.json [hunk 2]
```diff
@@ -0,0 +1,980 @@
...
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 3]
```diff
@@ -0,0 +1,980 @@
...
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 4]
```diff
@@ -0,0 +1,980 @@
...
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 5]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": "50"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": true
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 6]
```diff
@@ -0,0 +1,980 @@
...
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": "35",
+              "version_removed": "46"
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Please copy values from the Opera entry.

File: api/MediaStreamTrack.json [hunk 7]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/onmute",
+          "support": {
+            "webview_android": {
+              "version_added": true
+            },
+            "chrome": {
+              "version_added": true
+            },
+            "chrome_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 8]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/onunmute",
+          "support": {
+            "webview_android": {
+              "version_added": true
+            },
+            "chrome": {
+              "version_added": true
+            },
+            "chrome_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 9]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/onoverconstrained",
+          "support": {
+            "webview_android": {
+              "version_added": true
+            },
+            "chrome": {
+              "version_added": true
+            },
+            "chrome_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 10]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/onended",
+          "support": {
+            "webview_android": {
+              "version_added": true
+            },
+            "chrome": {
+              "version_added": true
+            },
+            "chrome_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 11]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": false
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": true
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: true

File: api/MediaStreamTrack.json [hunk 12]
```diff
@@ -0,0 +1,980 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "clone": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/clone",
+          "support": {
+            "webview_android": {
+              "version_added": "48"
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Actually, it was version 45. 

Here's the commit: https://chromium.googlesource.com/chromium/src/+/776e9a66223e95141b0b8bc6c6f9a847cd5fd8c4

Here's the lookup that tells when the commit landed: https://storage.googleapis.com/chromium-find-releases-static/776.html#776e9a66223e95141b0b8bc6c6f9a847cd5fd8c4

File: api/MediaStreamTrack.json [hunk 13]
```diff
@@ -0,0 +1,980 @@
...
+            },
+            "chrome": {
+              "version_added": true
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: 45

File: api/MediaStreamTrack.json [hunk 14]
```diff
@@ -0,0 +1,980 @@
...
+            },
+            "chrome_android": {
+              "version_added": true
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: 45

File: api/MediaStreamTrack.json [hunk 15]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": null
+            },
+            "ie": {
+              "version_added": null
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: 32

File: api/MediaStreamTrack.json [hunk 16]
```diff
@@ -0,0 +1,980 @@
...
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: 32

File: api/MediaStreamTrack.json [hunk 17]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/getCapabilities",
+          "support": {
+            "webview_android": {
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: All three of these should be `true`.

File: api/MediaStreamTrack.json [hunk 18]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": null
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": null
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Both should be `true`.

File: api/MediaStreamTrack.json [hunk 19]
```diff
@@ -0,0 +1,980 @@
...
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Both should be true

File: api/MediaStreamTrack.json [hunk 20]
```diff
@@ -0,0 +1,980 @@
...
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/MediaStreamTrack/stop",
+          "support": {
+            "webview_android": {
+              "version_added": "61"
+            },
+            "chrome": {
+              "version_added": true
+            },
+            "chrome_android": {
+              "version_added": true
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: All three should be 61.

File: api/MediaStreamTrack.json [hunk 21]
```diff
@@ -0,0 +1,980 @@
...
+              "version_added": false
+            },
+            "nodejs": {
+              "version_added": null
+            },
+            "opera": {
+              "version_added": null
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/MediaStreamTrack.json]: Both should be 48.

**Comment by jpmedley** [comment]
@foolip You're right. As you said to me elsewhere, `getSources()` is missing.

**Comment by foolip** [comment]
Given that, I wonder what the methodology for creating this file was, and if all other information in the current table has been preserved. @mec, can you describe the process?

**Comment by jpmedley** [comment]
I've been building them by hand using this: http://jsoneditoronline.org/#/

I usually paste in an existing file as a starting point then do a big chunk of the work with copy, paste, and delete.

**Comment by mec** [comment]
@foolip You've randomly come across the most complicated case I've attempted!

It's been quite hard as I am no expert in the API, merely trying to help out, generally I've been taking the page off the spreadsheet:

https://docs.google.com/spreadsheets/d/1ivgyPBr9Lj3Wvj5kyndT1rgGbX-pGggrxuMtrgcOmjM/edit#gid=926663640

And creating the top level json object. Then I put the compat data from the basic support entry in that object.

I then step through each linked method, property, handlers, etc. When there is a page with compat or other data, I add it. If there is no page, I leave it off as I have no way of migrating data. It looks like  I need to also scan the top level for missing or conflicting compat data.

If someone can advise on what to do in a case like this or if someone can suggest a best practice for this that would be super helpful.

**Comment by mec** [comment]
I've put @jpmedley changes in and added all the missing API with null values where I have nothing to go on, this should be in better shape now, I hope!

**Comment by foolip** [comment]
@Elchi3, is there a less manual process for conversion that some of you are using, or is eyeballing differences the only way here?

**Comment by Elchi3** [comment]
I'm not aware of an automated way from the old MDN compat tables into this data structure. The legacy tables are imperfect in terms of their structure and their data points as MDN is a free from HTML wiki everyone can edit. With the conversation we're standardizing the information now and we've issued a call to action displayed on all legacy tables, so that the community can help us with this effort. Discussion in PRs trump the data points in the old tables, as Joe or others know better than the old tables. 
This also has a good effect on the building a community around this repository. We're about 62% done and many people are involved now. When the migration is over, I hope we'll (collectively) focus on automation / tools / workflows that helps us verifying the data. MDN docs also don't cover all web features, so we will then also see how we go about adding more features that aren't yet on MDN (thus weren't part of the migration project). Does that answer your question? Happy to talk more.

**Pull request merged**